apps/relayer/README.md

```diff
@@ -1,5 +1,146 @@
-# Relayer service
+# MACI Relayer Service
 
-## Instructions
+This repository contains the setup and instructions to run the MACI Relayer service along with MongoDB using Docker.
 
-1. Add `.env` file (see `.env.example`).
+## Flow Diagram
+
+Below is a high-level diagram illustrating the flow of the Relayer Service:
+
+![MACI Relayer Flow](./relayer-diagram.png)
+
+## High-Level Overview of the Relayer:
+
+### Functionality
+
+The Relayer serves as an intermediary between users and the MACI system. It performs the following tasks:
+
+- Allows users to send messages without having to pay gas fees by processing these messages off-chain.
+- Receives user messages, such as votes or sign-ups.
+- Prepare messages batch and upload it to IPFS
+- Submits the message hashes to the Poll contract with IPFS hash of the batch.
+
+This process ensures that user actions are recorded on-chain without compromising their privacy.
+
+### Privacy Preservation
+
+By handling the creation and submission of zero-knowledge proofs, the Relayer ensures that sensitive user information remains confidential. This design choice is essential for maintaining the integrity and trustworthiness of the voting process.
```

Comment:
don't think the services handles creating any proofs, users generate their own proofs client side to prove they joined a poll